### PR TITLE
New features

### PR DESCRIPTION
**Pull request by ToniMaciel**
With this PR, we intend to add the following new features to OSean:

- Safe checkout for serialized objects, preventing losing information
- Serialization in Gradle projects
- Generates jars containing only tests of merge commit
- New CLI
- Solve testability transformations and build manager issues automatically
- Makes serialization optional

Also, we made the test suite more robust and solved issues like serializing duplicate objects, serializing beyond the time budget, and some instrumentations problems.

### CODE REVIEW

File: src/main/java/org/cli/CommandLineParametersParser.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+import org.util.input.TransformationOption;
+
+public class CommandLineParametersParser {
+  private final Options options;
+  private final CommandLineParser parser;
+  private final HelpFormatter formatter;
+
+  private List<Option> requiredOptions = new ArrayList<>();
+
+  public CommandLineParametersParser() {
```
Comment (pauloborba) [inline src/main/java/org/cli/CommandLineParametersParser.java:27]: essas opções estão atualizadas no README?

**Review by pauloborba — APPROVED** [review]

**Review by jpedroh — APPROVED** [review]

File: src/main/java/org/cli/CommandLineParametersParser.java (hunk 1 above)
Comment (jpedroh) [reply src/main/java/org/cli/CommandLineParametersParser.java:27]: Estão sim!

Comment (ToniMaciel) [reply src/main/java/org/cli/CommandLineParametersParser.java:27]: Havia esquecido de colocar a da opção de não serializar. Estava no help da ferramenta, mas não no README

**Review by jpedroh — APPROVED** [review]

**Review by jpedroh — APPROVED** [review]

**Review by leusonmario — APPROVED** [review]
